pinot-core/src/main/java/org/apache/pinot/core/startree/StarTreeUtils.java

```diff
@@ -179,4 +184,96 @@ public static boolean isFitForStarTree(StarTreeV2Metadata starTreeV2Metadata,
     // Check predicate columns
     return starTreeDimensions.containsAll(predicateColumns);
   }
+
+  /**
+   * Evaluates a given filter to ascertain if the OR clause is valid for StarTree processing.
+   *
+   * StarTree supports OR predicates on a single dimension only (d1 < 10 OR d1 > 50).
+   *
+   * @return Pair of the result and the single literal on which the predicate is based. If the
+   * predicate is not valid for execution on StarTree, literal value will be null
+   */
+  private static Pair<Boolean, String> isOrClauseValidForStarTree(FilterContext filterContext) {
+    assert filterContext != null;
+
+    Set<String> seenLiterals = new HashSet<>();
+
+    if (!isOrClauseValidForStarTreeInternal(filterContext, seenLiterals)) {
+      return Pair.of(false, null);
+    }
+
+    boolean result = seenLiterals.size() == 1;
+
+    return Pair.of(result, result ? seenLiterals.iterator().next() : null);
+  }
+
+  /** Internal processor for the above evaluator */
+  private static boolean isOrClauseValidForStarTreeInternal(FilterContext filterContext, Set<String> seenLiterals) {
+    assert filterContext != null;
+
+    if (filterContext.getType() == FilterContext.Type.OR) {
+      List<FilterContext> childFilterContexts = filterContext.getChildren();
+
+      for (FilterContext childFilterContext : childFilterContexts) {
+        isOrClauseValidForStarTreeInternal(childFilterContext, seenLiterals);
+      }
```

Comment:
Return `true` here

Reply:
We return true right after this if block -- should that suffice?